content/blog/2017/2017-01-01-conditional.adoc

```diff
@@ -0,0 +1,309 @@
+            choices: ['greeting','silence'].join('\n'),
+            description: '',
+            name: 'REQUESTED_ACTION')
+    }
+
+    stages {
+        stage ('Speak') {
+            when {
+                // Only say hello if a "greeting" is requested
+                expression { return params.REQUESTED_ACTION == 'greeting' }
+            }
+            steps {
+                echo "Hello, bitwiseman!"
+            }
+        }
+    }
+}
+// Script //
+properties ([
+    parameters ([
+        choice (
+            // choices must be newline separated
+            choices: ['greeting','silence'].join('\n'),
+            description: '',
+            name : 'REQUESTED_ACTION')
+    ])
+])
+
+node {
+    stage ('Speak') {
+        // Only say hello if a "greeting" is requested
+        if (params.REQUESTED_ACTION == 'greeting') {
+            echo "Hello, bitwiseman!"
+        }
+    }
+}
+----
+
+Here's the output when I run this project with `REQUESTED_ACTION` set to "greeting":
+
+[source]
+----
+[Pipeline] node
+Running on osx_mbp in /Users/bitwiseman/jenkins/agents/osx_mbp/workspace/pipeline-conditional
+[Pipeline] {
+[Pipeline] stage
+[Pipeline] { (Speak)
+[Pipeline] echo
+Hello, bitwiseman!
+[Pipeline] }
+[Pipeline] // stage
+[Pipeline] }
+[Pipeline] // node
+[Pipeline] End of Pipeline
+Finished: SUCCESS
+----
+
+When at I pass the value "silence", the only change is "Hello, bitwiseman!" is not printed.
+
+The Pipeline code is considerably more compact than the Jenkins UI presentation.
+I had to do a some web searching to find out how to had build
+parameters to Pipeline
+(link:https://issues.jenkins-ci.org/browse/WEBSITE-281[One more task] for our
+documentation efforts).
+It took several iterations to get this pipeline to work, but it was pretty quick.
+
+
+=== Conditions
+The previous example showed the "Strings match" condition and it's Pipeline equivalent.
+Let's look at couple more interesting conditions and their pipeline equivalents.
+
+==== Boolean condition
+
+[QUOTE]
+====
+Expand the Token Macro and run the build step if it evaluates to true.
+
+Truth is a case insensitive match of one of the following:
+1 (the number one), Y, YES, T, TRUE, ON or RUN.
+====
+
+You might think that a boolean condition would be the simplest condition, but it isn't.
+Since it works with string values from tokens, the Conditional BuildStep plugin offers
+a number of ways to indicate true or false.
+Pipeline can duplicate these, but we might consider if a simpler expression
+would suffice on a case-by-case basis.
+
+.Pipeline
+[pipeline]
+----
+// Declarative //
+when {
+    // case insensitive regular expression for truthy values
+    expression { return token ==~ /(?i)(Y|YES|T|TRUE|ON|RUN)/ }
+}
+steps {
+    /* step */
+}
+
+// Script //
+// case insensitive regular expression for truthy values
+if (token ==~ /(?i)(Y|YES|T|TRUE|ON|RUN)/) {
+    /* step */
+}
+----
+
+==== Logical "OR" of conditions
+
+[QUOTE]
```

Comment:
Because it is basically a quote from the conditional buildstep plugin description for this condition.